backend/src/server.js

```diff
@@ -11,56 +13,38 @@ const fastify = require('fastify')({
     }
 });
 
-/**
- * CORS options.
- */
-fastify.register(fastifyCors, { 
-    /**
-     * Put all the api options in one place.
-     */
-});
-
-/**
- * Fastify global hooks allow us to hook into the pre-
- * request handler. This is one of the many options we have
- * for authorizing responses.
- * 
- * @link https://www.fastify.io/docs/latest/Hooks/
- */
-fastify.addHook('preHandler', async (request, reply) => {
-    request.log.info('I  am running on each request');
-    const isAuth = await authPreHandler(false);
-    if ( isAuth ) {
-        reply
-        .code(401)
-        .header('Content-Type', 'application/json; charset=utf-8')
-        .send({ message: "Not authorized." });
-    } else {
-        reply.code(200);
-    }
-});
-
-// Register root route.
-fastify.route({
-  method: 'GET',
-  url: '/',
-  handler: async () => {
-    return { hello: 'world' }
-  }
-});
+// Fastify server config.
+fastify
+  .decorate('verifyJWT', (req, res, done) => {
```

Comment:
wish there were comments here. though its kinda self-documenting, its a bit cryptic still, imo

Reply:
Yeah, agreed and I'll see what I can do. Because of the chaining nature of the fastify API, it doesn't lend itself well to documentation as far as each method goes(unless I were to write a novel on line 16)